spec.md

```diff
@@ -586,18 +589,68 @@ schema being used.
           "40 concurrent connections"
         ]
       }
+    }, {
+      "name": "fake-deprecated-plan",
+      "id": "009aba-XXXX-XXXX-XXXX-dace631cd648",
+      "description": "This is deprecated.",
+      "free": false,
+      "deprecation": {
+        "description": "Don't use this anymore.",
+        "since": "2018-04-20T17:00:00+00:00",
+        "eol": "2020-02-28T08:00:00+00:00",
+        "alternatives": [{
+            "description": "Use this plan instead.",
+            "plan_id": "0f4008b5-XXXX-XXXX-XXXX-dace631cd648"
+          }
+        ]
+      }
     }]
   }]
 }
 ```
 
+##### Deprecation Object 
+
+| Response Field | Type | Description |
+| --- | --- | --- |
+| description | string | A short description of the deprecation. |
+| since | date string | Date and time when the Service Broker deprecated the service or plan. If specified, it is expected to be an ISO 8601 formatted date and time, with time zone. |
+| eol | date string | Date and time when the Service Broker expects to end-of-life (remove) the service or plan from the catalog. If specified, it is expected to be an ISO 8601 formatted date and time, with time zone. |
+| alternatives | array of [CatalogObjectId](#catalog-object-id-object) objects | An array of services or plans that the Service Broker considers to be good alternatives. |
+
+\* Fields with an asterisk are REQUIRED.
+
+##### Catalog Object Id Object 
+
+| Response Field | Type | Description |
```

Comment:
Something should be required in this. No idea what it would mean to have an empty one.

Reply:
I thought of this, it could be that the broker just wants to pass a description that just says "no alternative. go away"

Reply:
no alternative is ok, but we should have something in here, even a start date. If the entire struct is empty then why have it at all. And if we don't have it then the referencing property is empty - and that just seems silly.

Reply:
yea, because then we enter the rathole of adding text to makes sure people don't do what they might naturally want to do like add `omitempty` on things.  In many cases empty == nil == not there. Trying to draw a distinction between those, while not impossible, is more trouble than its worth. I don't think its a burden to ask for at least one of these fields to be there.

Reply:
I disagree, I think it is fine to have something like this:

```
{
      "name": "fake-deprecated-plan",
      "id": "009aba-XXXX-XXXX-XXXX-dace631cd648",
      "description": "This is deprecated.",
      "free": false,
      "deprecation": {}
        ]
      }
    }
```

Reply:
I have made description be required and non empty string. So lazy broker will do this:

```
{
      "name": "fake-deprecated-plan",
      "id": "009aba-XXXX-XXXX-XXXX-dace631cd648",
      "description": "This is deprecated.",
      "free": false,
      "deprecation": {
         "description": "deprecated"
      }
}
```

Reply:
Also added MUST around service_id or plan_id is needed for the alternatives object to be valid.

Reply:
So we are saying service brokers MUST provide alternatives? What if they are actually deprecating a service entirely?